File: backend/app/db/models/mart_bill_item.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, Numeric, String
from sqlalchemy.orm import relationship

from .base_class import Base
from .mixins import AuditMixin


class MartBillItem(Base, AuditMixin):
    __tablename__ = "invoice_item"

    id = Column(Integer, primary_key=True, index=True)
    invoice_id = Column(Integer, ForeignKey("invoice.id"), nullable=False)
    item_id = Column(Integer, ForeignKey("item.id"), nullable=True)
    warehouse_id = Column(
        Integer, ForeignKey("warehouse.id"), nullable=False, index=True
    )

    # fields from your parser
    hsn_code = Column(String, nullable=True)
    item_code = Column(String, nullable=True)
    item_name = Column(String, nullable=False)
    quantity = Column(Numeric(10, 3), nullable=False)
    uom = Column(String, nullable=False)
    price = Column(Numeric(18, 6), nullable=False)
    total = Column(Numeric(18, 6), nullable=False)
    invoice_date = Column(DateTime, nullable=False)
    store_name = Column(String, nullable=False)

    bill = relationship("MartBill", back_populates="items")
    item = relationship("Item")
    warehouse = relationship("Warehouse")

File: mobile/lib/models/mart_bill_item.dart
import 'json_parsers.dart';

class MartBillItem {
  final int id;
  final int invoiceId;
  final int? itemId;
  final String resolutionStatus;
  final String itemName;
  final String? itemCode;
  final String? storeName;
  final double quantity;
  final String uom;
  final double price;
  final double total;

  MartBillItem({
    required this.id,
    required this.invoiceId,
    this.itemId,
    required this.resolutionStatus,
    required this.itemName,
    this.itemCode,
    this.storeName,
    required this.quantity,
    required this.uom,
    required this.price,
    required this.total,
  });

  bool get isUnresolved => resolutionStatus == 'UNRESOLVED';

  factory MartBillItem.fromJson(Map<String, dynamic> json) {
    return MartBillItem(
      id: JsonParsers.parseInt(json['id']),
      invoiceId: JsonParsers.parseInt(json['invoice_id']),
      itemId: json['item_id'] as int?,
      resolutionStatus: json['resolution_status'] ?? 'UNRESOLVED',
      itemName: json['item_name'] ?? '',
      itemCode: json['item_code'],
      storeName: json['store_name'],
      quantity: JsonParsers.parseNum(json['quantity']),
      uom: json['uom'] ?? '',
      price: JsonParsers.parseNum(json['price']),
      total: JsonParsers.parseNum(json['total']),
    );
  }
}
